ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,390 @@
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(
+        self, param_name: str, save_history: bool = False,
+    ):
+        super(StateParamScheduler, self).__init__(param_name, save_history)
+
+    def attach(
+        self,
+        engine: Engine,
+        event: Union[str, Events, CallableEventWithFilter, EventsList] = Events.ITERATION_COMPLETED,
+    ) -> None:
+        """Attach the handler to the engine. Once the handler is attached, the ``Engine.state`` will have a new
+        attribute with the name ``param_name``. Then the current value of the parameter can be retrieved from
+        ``Engine.state`` when the engine is running.
+
+        Args:
+            engine: trainer to which the handler will be attached.
+            event: trigger ``param_name`` value update.
+
+        """
+        if hasattr(engine.state, self.param_name):
+            raise ValueError(
+                f"Attribute: '{self.param_name}' is already defined in the Engine.state."
+                f"This may be a conflict between multiple StateParameterScheduler handlers."
+                f"Please choose another name."
+            )
+
+        if self.save_history:
+            if not hasattr(engine.state, "param_history") or engine.state.param_history is None:  # type: ignore
+                setattr(engine.state, "param_history", {})
+                engine.state.param_history.setdefault(self.param_name, [])  # type: ignore[attr-defined]
+
+        engine.add_event_handler(event, self)
+
+    def __call__(self, engine: Engine) -> None:
+        self.event_index += 1
+        value = self.get_param()
+        setattr(engine.state, self.param_name, value)
+        if self.save_history:
+            engine.state.param_history[self.param_name].append(value)  # type: ignore[attr-defined]
+
+    @classmethod
+    def simulate_values(cls, num_events: int, **scheduler_kwargs: Any) -> List[List[int]]:
+        """Method to simulate scheduled engine state parameter values during `num_events` events.
+
+        Args:
+            num_events: number of events during the simulation.
+            scheduler_kwargs: parameter scheduler configuration kwargs.
+
+        Returns:
+            event_index, value
+
+        Examples:
+
+        .. code-block:: python
+
+            import matplotlib.pyplot as plt
+            import numpy as np
+
+            step_state_param_values = np.array(
+                StepStateScheduler.simulate_values(
+                    num_events=20, param_name="step_scheduled_param", initial_value=10, gamma=0.99, step_size=5
+                )
+            )
+
+            plt.plot(step_state_param_values[:, 0], step_state_param_values[:, 1], label="learning rate")
+            plt.xlabel("events")
+            plt.ylabel("values")
+            plt.legend()
+
+        """
+        keys_to_remove = ["save_history"]
+        for key in keys_to_remove:
+            if key in scheduler_kwargs:
+                del scheduler_kwargs[key]
```

Comment:
@fco-dv Could we test this part ?

Reply:
yes, but what do you have in mind precisely :) ?